Representable/representable
# Form Validation Updates

Closes #103. Closes #111. Closes #105.

A few form validation changes:

- If the user enters a polygon correctly but makes a mistake in another field the map will redraw the polygon, highlight the right blocks, and zoom into the right state. Uses sessionStorage to remember the state found by the user via the geocoder.
- Added form entry errors on all fields using alerts.
- Added red borders to fields in order highlight fields missed by the user.
- FIXES error on pressing "Draw" again. In the past, if user pressed draw, the polygon would disappear but that wouldn't update the entry. Thus, it was possible for the user to submit the wrong polygon if they didn't pay attention.

How to test:

- Try to leave a field non filled. The border should turn red.
- Draw a polygon but DONT fill any of the interests fields. This will trigger an error but will attempt to submit the form. Then you can observe the map behavior.
- Draw a polygon. Then press on "Draw polygon" again. Try to submit form. It should not let you.

One screenshot (hard to replicate other behavior):
<img width="1074" alt="Screen Shot 2020-05-25 at 23 21 19" src="https://user-images.githubusercontent.com/8714338/82858522-1d530900-9ee2-11ea-95fe-e4377d363cf6.png">
